llama_index/embeddings/openai.py

```diff
@@ -103,6 +104,13 @@ class OpenAIEmbeddingModeModel(str, Enum):
 @retry(
     wait=wait_random_exponential(min=1, max=20),
     stop=stop_all(stop_after_attempt(6), stop_after_delay(60)),
+    retry=(
+        retry_if_exception_type(openai.error.Timeout)
+        | retry_if_exception_type(openai.error.APIError)
+        | retry_if_exception_type(openai.error.APIConnectionError)
```

Comment:
Does `tenacity` not support passing a list of retryable error types, instead of repeating `retry_if_exception_type` many times?

I am wondering if this is supported:

```python
retry_if_exception_type((openai.error.Timeout, openai.error.APIError, openai.error.APIConnectionError, openai.error.RateLimitError, openai.error.ServiceUnavailableError))
```

Reply:
<img width="618" alt="Screenshot 2023-10-24 at 6 30 38 PM" src="https://github.com/run-llama/llama_index/assets/108733252/5938050d-f4df-4d45-a7aa-810c3f98acae">

According to the source code of the `retry_if_exception_type` class, yep @jamesbraza I think you are correct we can pass in a tuple of error types directly instead of repeating retry_if_exception_type many times. Will update this logic in `_create_retry_decorator` directly and use that so that it can be a singular util for all openai retry mechanisms.